### PR TITLE
[RSDK-1954] add slam stream apis

### PR DESCRIPTION
**Pull request by nicksanford**
Ticket: https://viam.atlassian.net/browse/RSDK-1954

- Add GetInternalStateStream & GetPointCloudMapStream server streaming grpc endpoints to the slam service proto.
- Add associated request/response message types (required in order to conform to linting rules around grpc method requiest / response type names).
- Add comment describing unary GetInternalState & GetPointCloudMap grpc endpoints as deprecated.

### CODE REVIEW

File: proto/viam/service/slam/v1/slam.proto
Comment (nicksanford) [inline proto/viam/service/slam/v1/slam.proto:162]: Given that each `GetPointCloudMapStreamResponse`  messages will now only contain a chunk of the PCD file, I changed the names of this message member to include `_chunk` in order to prevent a caller from thinking that it contains an entire PCD file.

File: proto/viam/service/slam/v1/slam.proto
Comment (nicksanford) [inline proto/viam/service/slam/v1/slam.proto:173]: I added the `_chunk` suffix to this member for the same reasons mentioned here: https://github.com/viamrobotics/api/pull/196/files#r1108841266

Comment (jeremyhyde-viam) [reply proto/viam/service/slam/v1/slam.proto:173]: Good idea, can you mention it in the doc you are creating that this additional API change exists? I don't think we need to hold off on starting but I want to make sure Tess is aware.

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [reply proto/viam/service/slam/v1/slam.proto:162]: Will `GetPointCloudMapStreamResponse` be the final name or will this be reverted to `GetPointCloudMapResponse` down the line?

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [inline proto/viam/service/slam/v1/slam.proto:153]: Not necessary to have anew request, can use the old one. Let us just be consistent with the final naming convention we wish to adopt for the Response (i.e. name contains stream or no)

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [inline proto/viam/service/slam/v1/slam.proto:159]:
```suggestion
  // Pointclouds are returned in chunks as a set of bytes which can be
```

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [inline proto/viam/service/slam/v1/slam.proto:160]:
```suggestion
  // aggregated into the standard PCD format
```

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [inline proto/viam/service/slam/v1/slam.proto:67]:
```suggestion
  // TODO (RSDK-1066): This will be renamed to GetInternalState
```

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [inline proto/viam/service/slam/v1/slam.proto:58]:
```suggestion
  // TODO (RSDK-1066): This will soon be renamed to GetPointCloudMap
```

**Review by jeremyhyde-viam — APPROVED** [review]
Approved, one or two small questions/comments regarding naming but not blockers

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [reply proto/viam/service/slam/v1/slam.proto:173]: I like making this explicitly know that the message bytes are chunks and not the full PCD

**Review by JohnN193 — APPROVED** [review]

File: proto/viam/service/slam/v1/slam.proto
Comment (nicksanford) [reply proto/viam/service/slam/v1/slam.proto:153]: Unfortunately we need to create a new request & response type b/c `make lint` will fail the build if I try to reuse `GetPointCloudMapRequest`. Same goes for the other request / response message types.

Also re: https://github.com/viamrobotics/api/pull/196#discussion_r1108858538 (github won't let me comment on it for some reason), answer is yes.

File: proto/viam/service/slam/v1/slam.proto
Comment (nicksanford) [reply proto/viam/service/slam/v1/slam.proto:162]: We will change the name to `GetPointCloudMapResponse`.

File: proto/viam/service/slam/v1/slam.proto
Comment (jeremyhyde-viam) [reply proto/viam/service/slam/v1/slam.proto:153]: Got it. So during RSDK-1066 where we will be renaming and removing old functions, we will end up with:
- GetPointCloudMapRequest
- GetPointCloudMapResponse
- GetPointCloudMap
- GetInternalStateRequest
- GetInternalStateResponse
- GetInternalStateMap

Comment (nicksanford) [reply proto/viam/service/slam/v1/slam.proto:153]: yes

File: proto/viam/service/slam/v1/slam.proto
Comment (edaniels) [inline proto/viam/service/slam/v1/slam.proto:39]: I'd try using this to have deprecation code generate https://stackoverflow.com/a/43380742

File: proto/viam/service/slam/v1/slam.proto
Comment (edaniels) [inline proto/viam/service/slam/v1/slam.proto:60]:
```suggestion
  // GetPointCloudMap returns the latest point cloud map available
```

SLAM library is an impl detail. Same applies to other comments

File: proto/viam/service/slam/v1/slam.proto
Comment (edaniels) [inline proto/viam/service/slam/v1/slam.proto:68]:
```suggestion
  // GetInternalStateStream returns the internal map as defined by the specified slam
```

File: proto/viam/service/slam/v1/slam.proto
Comment (edaniels) [reply proto/viam/service/slam/v1/slam.proto:153]: it's verbose but it's consistent!

File: proto/viam/service/slam/v1/slam.proto
Comment (edaniels) [inline proto/viam/service/slam/v1/slam.proto:160]: should be a bit more specific in the wording. I think it may just be concatenating the responses from the beginning and end of the stream and nothing in between.

**Review by edaniels — CHANGES_REQUESTED** [review]

File: proto/viam/service/slam/v1/slam.proto
Comment (edaniels) [reply proto/viam/service/slam/v1/slam.proto:39]: If this doesn't work we can leave as is

**Review by edaniels — APPROVED** [review]
Also LGTM once changes are in. Don't wanna block you from merging

File: proto/viam/service/slam/v1/slam.proto
Comment (nicksanford) [reply proto/viam/service/slam/v1/slam.proto]: Works great! Thanks for the suggestion.

**Pull request merged**
